code/game/objects/items/cigs_lighters.dm

```diff
@@ -226,7 +227,8 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 		qdel(src)
 		return
 	open_flame()
-	if(reagents && reagents.total_volume)
+	if((reagents && reagents.total_volume) && (nextdragtime <= world.time))
+		nextdragtime = world.time + dragtime

```

Comment:
tbh i'd like to see a little bit of randomization added to this instead of it just being a puff every minute and a half.

Reply:
Used to be random before #39334 which caused this issue of cigarettes dumping their reagents so fast as they do now, I'm not sure if removing that was part of fixing that bug or not

Reply:
Looking at #39334, the issue was that the reagent reaction itself had a 15% chance of happening.

What I meant was having the delay between puffs be slightly random, ie; `nextdragtime = world.time + (dragtime * rand(0.8, 1.2))`. This would make the time between puffs be randomized between 8 and 12 seconds (based off 100ds / 10 seconds as you have it now.)

Reply:
good idea